### ISSUE TITLE
Update translation and question about Italian

### ISSUE THREAD

**Issue by lachlan-00** [ampache/ampache#2784]
@Psy-Virus could you do a sync of develop soon? The big source changes branch has gone in and I've tried to clean up /prep the strings as much as possible so now would be a good time for a sync

There was also Elettriko in the telegram group trying to get into the Italian group to translate but wasn't able to. Has it been locked?

**Comment by usox** [comment]
Annnd to add another question: which tools are used for extraction of the translations keys? For the future refactoring in the template section, it's necessary to extract translation keys from xhtml files.

**Comment by Psy-Virus** [comment]
> @Psy-Virus could you do a sync of develop soon? The big source changes branch has gone in and I've tried to clean up /prep the strings as much as possible so now would be a good time for a sync

Wanted to do it yesterday but then I saw all the changes and Ampache just didn't want to do anything anymore XD
I'll try to get the translations done in the next hours!

> There was also Elettriko in the telegram group trying to get into the Italian group to translate but wasn't able to. Has it been locked?

I guess he/she was just impatient. For spam protection, I have to accept translators manually. And that's done when I'm at home. I got the email at 23:40 / 11:40 pm on Monday. That's usually sleepy sleepy time for me ^^ And I didn't check the mails until ~4+ pm the next day because there's also worky wrorky time :D
I'd love to have the luxury of having home office. But customers probably would freak out if I send a robot to build or fix their electrical installation. XDD


> Annnd to add another question: which tools are used for extraction of the translations keys? For the future refactoring in the template section, it's necessary to extract translation keys from xhtml files.

You really mean xhtml and not xml? XHTML just doesn't really make sense to me right now :D
But I'll update the gather-messages.sh to everything you want! ;)

Basically this is the tool I use in conjunction with xgettext and the mighty bash ofc ^^
I just checked the history of the file... I can't even remember changing so much on it ... weird O.o





:)

**Comment by usox** [comment]
> You really mean xhtml and not xml? XHTML just doesn't really make sense to me right now :D
> But I'll update the gather-messages.sh to everything you want! ;)
> 
> Basically this is the tool I use in conjunction with xgettext and the mighty bash ofc ^^
> I just checked the history of the file... I can't even remember changing so much on it ... weird O.o

We have xhtml files, but yes: basically xml - like in here: https://github.com/ampache/ampache/blob/develop/resources/templates/stats.xhtml

I can provide some shell+php code to extract the translation keys from those templates, you will just need to merge them with the translation keys extracted from the php source.

**Comment by Psy-Virus** [comment]
So much has changed... It's a real pity that I just don't have the time anymore to really follow Ampache. I really miss it, as this project motivated me to really learn to program. 😢

> I can provide some shell+php code to extract the translation keys from those templates, you will just need to merge them with the translation keys extracted from the php source.

This shouldn't be a problem. But maybe the time has also come to finally automate the gather-translations process...
As @lachlan-00 has more or less overtaken the reigns and has a server infrastructure (Yeah, I read the wiki :P) this could be done pretty easy I guess.

**Comment by usox** [comment]
@Psy-Virus Ha, just forgot this thread.

I18NFool (https://metacpan.org/pod/I18NFool) can be used to extract the translations keys from the templates.
`i18nfool-extract resources/templates/*.xhtml`

If necessary, it should be possible to provide some kind of php implementation of this extractor, though.
